### DISCUSSION TITLE
`or`-patterns (tweet)

### DISCUSSION THREAD

**Discussion by ggreif** [dfinity/motoko#4822] [Show and tell]
`or`-patterns let you collapse two (or more) `case` branches into one under certain circumstances.

The patterns are permitted to bind variables as long as the same variables are bound with identical types.
``` Motoko
switch animal { // NOT ACTUAL ADVICE!
  case (#wolf or #bear(#brown)) run();
  case (#deer or #bear(#black)) freeze();
  case (#snake or #scorpion) applyAntivenom();
}
```

**Comment by jessiemongeon1** [comment]
Schedule for Friday 7/20
